sml/src/sml_time.c

```diff
@@ -38,18 +38,29 @@ sml_time *sml_time_parse(sml_buffer *buf) {
 
 	sml_time *tme = sml_time_init();
 
-	if (sml_buf_get_next_type(buf) != SML_TYPE_LIST) {
-		buf->error = 1;
-		goto error;
-	}
+    // workaround Holley DTZ541
+    // if SML_ListEntry valTime (SML_Time) is given there are missing bytes:
+    // 0x72: indicate a list for SML_Time with 2 entries
+    // 0x62 0x01: indicate secIndex
+    // instead, the DTZ541 starts with 0x65 + 4 bytes secIndex
+    // the workaround will add this information during parsing
+    if (sml_buf_get_current_byte(buf) == (SML_TYPE_UNSIGNED | 5)) {
+        u8 *tag = malloc(sizeof(u8));
```

Comment:
this code has no effect. tag is a local variable that runs out of scope next line... (only impact: memory leak)
you might want to set
```
tme->tag = SML_TIME_SEC_INDEX;
```
?

Reply:
thought `tme->tag` is a pointer to a `u8`. Thats why i allocated the byte here.
`tme->tag` will be freed in `line 108: sml_number_free(tme->tag)`.

please correct me if iam wrong on that one

Reply:
hmm. I didn't check the type from tml->tag. But in any case you need to assign it then e.g.
```
tme->tag = malloc(...) or
tme->tag = tag;
```

Reply:
ah, now i can see it :D 
good point!